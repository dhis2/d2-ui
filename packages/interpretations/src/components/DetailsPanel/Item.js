import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import styles from './styles/Details.style';

export const Item = ({ classes, label, text }) => (
    <div className={classes.detailsCardItem}>
        {label && <label className={classes.item}>{label}:</label>}
        {text}
    </div>
);

Item.propTypes =Â {
    classes: PropTypes.object.isRequired,
    label: PropTypes.string,
    text: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
};

export default withStyles(styles)(Item);
