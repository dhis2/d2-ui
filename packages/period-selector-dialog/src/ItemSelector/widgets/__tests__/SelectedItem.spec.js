import React from 'react'
import { shallow } from 'enzyme'
import Item from '../SelectedItem'

describe('Selected item component', () => {
    let props
    let shallowItem

    const item = () => {
        if (!shallowItem) {
            shallowItem = shallow(<Item {...props} />)
        }
        return shallowItem
    }

    beforeEach(() => {
        props = {
            id: 'selected-item-test-id',
            name: 'I am a selected item',
            index: 0,
            highlighted: false,
            ghost: false,
            onClick: jest.fn(),
            onRemove: jest.fn(),
        }
        shallowItem = undefined
    })

    it('renders an unhighlighted item', () => {
        const wrapper = item()

        expect(wrapper).toMatchSnapshot()
    })

    it('renders a highlighted item', () => {
        props.highlighted = true
        const wrapper = item()

        expect(wrapper).toMatchSnapshot()
    })

    it('renders a ghost item', () => {
        props.ghost = true
        const wrapper = item()

        expect(wrapper).toMatchSnapshot()
    })

    describe('onClick', () => {
        it('fires onClick property', () => {
            item()
                .props()
                .onClick({ preventDefault: () => undefined })

            expect(props.onClick).toBeCalledTimes(1)
        })

        it('fires onClick with correct arguments when metaKey pressed', () => {
            item()
                .props()
                .onClick({
                    preventDefault: () => undefined,
                    metaKey: true,
                    ctrlKey: false,
                    shiftKey: false,
                })

            expect(props.onClick).toBeCalledTimes(1)
            expect(props.onClick).toBeCalledWith({
                isCtrlPressed: true,
                isShiftPressed: false,
                index: 0,
                id: props.id,
            })
        })

        it('fires onClick with correct arguments when ctrlKey pressed', () => {
            item()
                .props()
                .onClick({
                    preventDefault: () => undefined,
                    metaKey: false,
                    ctrlKey: true,
                    shiftKey: false,
                })

            expect(props.onClick).toBeCalledTimes(1)
            expect(props.onClick).toBeCalledWith({
                isCtrlPressed: true,
                isShiftPressed: false,
                index: 0,
                id: props.id,
            })
        })

        it('fires onClick with correct arguments when shiftKey pressed', () => {
            item()
                .props()
                .onClick({
                    preventDefault: () => undefined,
                    metaKey: false,
                    ctrlKey: false,
                    shiftKey: true,
                })

            expect(props.onClick).toBeCalledTimes(1)
            expect(props.onClick).toBeCalledWith({
                isCtrlPressed: false,
                isShiftPressed: true,
                index: 0,
                id: props.id,
            })
        })
    })
})
