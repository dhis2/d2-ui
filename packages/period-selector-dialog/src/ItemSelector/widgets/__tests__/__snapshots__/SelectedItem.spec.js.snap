// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selected item component renders a ghost item 1`] = `
<div
  className="item ghost selected-item"
  data-test="dimension-item-selected-item-test-id"
  onClick={[Function]}
>
  <ItemIcon
    backgroundColor="#00796b"
  />
  <span
    className="item-label"
  >
    I am a selected item
  </span>
  <DeselectIconButton
    fill="#00796b"
    onClick={[Function]}
  />
  <style />
</div>
`;

exports[`Selected item component renders a highlighted item 1`] = `
<div
  className="item highlighted-item"
  data-test="dimension-item-selected-item-test-id"
  onClick={[Function]}
>
  <ItemIcon
    backgroundColor="#ffffff"
  />
  <span
    className="item-label highlighted-text"
  >
    I am a selected item
  </span>
  <DeselectIconButton
    fill="#ffffff"
    onClick={[Function]}
  />
  <style />
</div>
`;

exports[`Selected item component renders an unhighlighted item 1`] = `
<div
  className="item selected-item"
  data-test="dimension-item-selected-item-test-id"
  onClick={[Function]}
>
  <ItemIcon
    backgroundColor="#00796b"
  />
  <span
    className="item-label"
  >
    I am a selected item
  </span>
  <DeselectIconButton
    fill="#00796b"
    onClick={[Function]}
  />
  <style />
</div>
`;
